File: Two Sum/two_sum.js
import { arr } from './array'

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    let numHash = {}
    // while loop
    console.time('loop')
    let i = 0
    while(true) {
        const currentNumber = nums.shift()
        if(numHash[target - currentNumber] !== undefined) {
            console.timeEnd('loop')
            return [numHash[target - currentNumber], i]
        }
        else {
            numHash[currentNumber] = i
            i++
        }
    }
}

console.log(twoSum(arr, 19999))

File: Two Sum/two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        x ={}
        for i in range(len(nums)):
            if(target - nums[i]) in x:
                return(x[target - nums[i]],i)
            else:
                x[nums[i]] = i

classInstance = Solution()
print(classInstance.twoSum(nums=[3,3],target=6))